src/CheckIn.js

```diff
@@ -69,14 +82,47 @@ class CheckIn extends React.Component {
     this.showInfo = this.showInfo.bind(this);
     this.renderActions = this.renderActions.bind(this);
     this.handleOptionsChange = this.handleOptionsChange.bind(this);
+    this.timer = undefined;
 
     this.state = {
       showPickers: false
     };
   }
 
-  componentDidMount() {
+  componentDidUpdate() {
+    const {
+      resources: {
+        checkinSettings,
+        scannedItems,
+      },
+      onSessionEnd,
+    } = this.props;
+
     this.focusInput();
+
+    if (this.timer) return;
```

Comment:
Please do not write one line if statements, always add curly braces.

Reply:
Hm, I like to write statements this way, and airbnb is suggesting it too. But I fixed it.